chia/full_node/coin_store.py

```diff
@@ -311,24 +314,25 @@ async def get_coin_states_by_puzzle_hashes(
             return []
 
         coins = set()
-        puzzle_hashes_db: Tuple[Any, ...]
-        if self.db_wrapper.db_version == 2:
-            puzzle_hashes_db = tuple(puzzle_hashes)
-        else:
-            puzzle_hashes_db = tuple([ph.hex() for ph in puzzle_hashes])
-        async with self.coin_record_db.execute(
-            f"SELECT confirmed_index, spent_index, coinbase, puzzle_hash, "
-            f"coin_parent, amount, timestamp FROM coin_record INDEXED BY coin_puzzle_hash "
-            f'WHERE puzzle_hash in ({"?," * (len(puzzle_hashes) - 1)}?) '
-            f"AND (confirmed_index>=? OR spent_index>=?)"
-            f"{'' if include_spent_coins else 'AND spent_index=0'}",
-            puzzle_hashes_db + (min_height, min_height),
-        ) as cursor:
+        for puzzles in chunks(puzzle_hashes, MAX_SQLITE_PARAMETERS):
+            puzzle_hashes_db: Tuple[Any, ...]
+            if self.db_wrapper.db_version == 2:
+                puzzle_hashes_db = tuple(puzzles)
+            else:
+                puzzle_hashes_db = tuple([ph.hex() for ph in puzzles])
+            async with self.coin_record_db.execute(
+                f"SELECT confirmed_index, spent_index, coinbase, puzzle_hash, "
+                f"coin_parent, amount, timestamp FROM coin_record INDEXED BY coin_puzzle_hash "
+                f'WHERE puzzle_hash in ({"?," * (len(puzzles) - 1)}?) '
+                f"AND (confirmed_index>=? OR spent_index>=?)"
+                f"{'' if include_spent_coins else 'AND spent_index=0'}",
+                puzzle_hashes_db + (min_height, min_height),
+            ) as cursor:
+
+                async for row in cursor:
```

Comment:
I think I like the middle ground of multiple `.fetchmany()` calls in general.  But, maybe in this form we just want to have the batching be based on the desired fetch batch size rather than the limit of the max query params?  Like maybe we want a batch size of 100 or something and go back to `.fetchall()`?

Reply:
I would like to keep it simple until we measure it to be too slow. It probably is a bit slower with an `async for` since it will (probably) need to check the future every turn through the loop, but I wouldn't think it makes a big difference.